test/controllers/EditSetControllerSpec.scala

```diff
@@ -48,868 +52,81 @@ class EditSetControllerSpec extends BaseSpec {
   "EditSetController GET /edit-set/{id}" should {
 
     "render the edit set page from a new instance of controller" in {
-      val messages: Map[String, Map[String, String]] =
-        Map(
-          "en" -> Map(
-            "edit-set.heading"       -> "Edit set: COAL 80 Sample",
-            "edit-set.table-caption" -> "Showing {0} - {1} of {2} records"
-          )
-        )
-      val mockMessagesApi = stubMessagesApi(messages)
-      val editSetInstance = inject[editSet]
-      val stub = stubControllerComponents()
+      val mockEditSet = mock[editSet]
+      val mockEditSetService = mock[EditSetService]
       val controller = new EditSetController(
-        DefaultMessagesControllerComponents(
-          new DefaultMessagesActionBuilderImpl(stubBodyParser(AnyContentAsEmpty), mockMessagesApi)(
-            stub.executionContext
-          ),
-          DefaultActionBuilder(stub.actionBuilder.parser)(stub.executionContext),
-          stub.parsers,
-          mockMessagesApi,
-          stub.langs,
-          stub.fileMimeTypes,
-          stub.executionContext
-        ),
-        editSetService,
-        editSetInstance
+        Helpers.stubMessagesControllerComponents(),
+        mockEditSetService,
+        mockEditSet
       )
-      val editSet = controller
-        .view("1")
-        .apply(
-          CSRFTokenHelper.addCSRFToken(
-            FakeRequest(GET, "/edit-set/1").withSession(SessionKeys.token -> validSessionToken)
-          )
+      when(mockEditSetService.get("1")).thenReturn(IO.pure(getEditSet()))
+      when(
+        mockEditSet(any[User], anyString(), anyString(), any[EditSetPage], any[EditSetEntryRowOrder])(
+          any[Messages],
+          any[Request[AnyContent]]
         )
-
-      status(editSet) mustBe OK
-      contentType(editSet) mustBe Some("text/html")
-      val document = asDocument(editSet)
-      document must haveCaption("Showing 1 - 10 of 12 records")
-      document must haveHeader("Edit set: COAL 80 Sample")
-    }
-
-    "render the edit set page from the application" in {
-      val controller = inject[EditSetController]
+      ).thenReturn(getEditSetDisplay)
       val editSet = controller
         .view("1")
         .apply(
           CSRFTokenHelper.addCSRFToken(
-            FakeRequest(GET, "/edit-set/1")
-              .withSession(SessionKeys.token -> validSessionToken)
+            FakeRequest(GET, "/edit-set/1").withSession(SessionKeys.token -> validSessionToken)
           )
         )
 
       status(editSet) mustBe OK
       contentType(editSet) mustBe Some("text/html")
```

Comment:
Do we want to assert the content type?

Reply:
@tna-erasmos No need. As mentioned above.

Reply:
Removed

Reply:
Confirmed.